server/resp/src/main/java/org/infinispan/server/resp/commands/list/blocking/AbstractBlockingPop.java

```diff
@@ -221,7 +223,12 @@ private PollListenerSynchronizer(EmbeddedMultimapListCache<byte[], byte[]> multi
          this.configuration = configuration;
          whenCompleteConsumer = (v, t) -> {
             if (t != null) {
-               resultFuture.completeExceptionally(t);
+               // Between installing the listener and doing the first poll, an entry was created.
+               // When trying to poll, we identify it is not a list element.
+               // We do not complete the future in this case, we leave it until the timeout elapses.
```

Comment:
Should we reregister the listener but with a reduced timeout in case if a valid value comes later?

Reply:
Do you mean to deregister the current one and register again with (current - start) timeout?

Initially, I thought of completing with `null`, which is what happens in the timeout. But after comparing with Redis, the caller remains blocked until a timeout or a list is created. I guess the current one is the simplest way to have the same behavior.

Reply:
Correct, reregister with (current - start) timeout so that if a list is added later it would be retrieved. Correct, me if I am wrong but the way this works now it will always cause a timeout if the type doesn't match, right?

Reply:
That's right. The listener should block until the timeout or a list is created. Not sure if showing the client console helps, but this is the expected behavior:

Client 1:
```text
127.0.0.1:6379> blpop list 0
1) "list"
2) "v3"
(25.54s)
```

Client 2:
```text
127.0.0.1:6379> set list v1
OK
127.0.0.1:6379> set list v2
OK
127.0.0.1:6379> del list
(integer) 1
127.0.0.1:6379> lpush list v1 v2 v3
(integer) 3
```

The command in client 1 is submitted first. The commands in client 2 are submitted later. The listener releases only after the push.

Reply:
Okay, I see how it all flows now. The indirect recursion of the polling makes it so there is no need to do a reregistration.